Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,91 +52,136 @@ async function main(): Promise<void> {
+                const wifRepo = new Repository(feedName, feedUrl, 'WIFbuild', token);
+                fs.appendFileSync(pypircPath, os.EOL + wifRepo.toString(), 'utf8');
+                federatedFeedSuccessCount++;
+            }
+            else {
+                throw new Error(tl.loc("Error_FailedToGetServiceConnectionAuth", entraWifServiceConnectionName));
+            }
+
+            return;
+        }
+        else if (entraWifServiceConnectionName || feedUrl) {
+            throw new Error(tl.loc("Error_MissingFeedUrlOrServiceConnection"));
+        }
+#endif
+
+        let reposList: string[] = [];
+
+        for (let entry of newEndpointsToAdd) {
+
+            let repo = new Repository(
+                entry.packageSource.feedName,
+                entry.packageSource.feedUri,
+                entry.username,
+                entry.password
+            );
 
-                let repo = new Repository(
-                    entry.packageSource.feedName,
-                    entry.packageSource.feedUri,
-                    entry.username,
-                    entry.password
-                );
+            if ((entry.packageSource.feedName in fileContent) || (usedRepos.has(repo.repository))) {
+                console.log(tl.loc("Warning_DuplicateEntryForFeed", entry.packageSource.feedName, repo.repository));
 
-                if (usedRepos.has(repo.repository)) {
-                    tl.warning(tl.loc("DuplicateRepoUrl", repo.repository));
-                    removeFromFeedCount(internalFeed, externalEndpoints, entry);
+                if (internalFeed.has(entry)) {
+                    internalFeed.delete(entry);
                     continue;
                 }
 
-                reposList.push(repo.toString() + `${os.EOL}`);
+                externalEndpoints.delete(entry);
+                continue;
+            }
+
+            tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
 
+            reposList.push(repo.toString() + `${os.EOL}`);
+            
+            (!fileContent.hasOwnProperty("distutils") || !fileContent["distutils"].hasOwnProperty("index-servers")) ?
+                fileContent["distutils"] = {["index-servers"]: `${entry.packageSource.feedName}`} : 
                 fileContent["distutils"]["index-servers"] += " " + entry.packageSource.feedName;
-            }
+        }
 
-            let encodedStr = ini.encode(fileContent);
-            fs.writeFileSync(pypircPath, encodedStr);
+        let encodedStr = ini.encode(fileContent);
+        fs.writeFileSync(pypircPath, encodedStr);
 
-            fs.appendFileSync(pypircPath, `${os.EOL}`, 'utf8');
+        fs.appendFileSync(pypircPath, `${os.EOL}`, 'utf8');
 
-            for (let repo of reposList) {
-                fs.appendFileSync(pypircPath, repo, 'utf8');
-            }
+        for (let repo of reposList) {
+            fs.appendFileSync(pypircPath, repo, 'utf8');
         }
+        //}
```

Comment:
commnet?

Reply:
The code was wrapped in an else statement before, I removed the comment to make the code cleaner